src/Cms/HasLock.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Data\Data;
+use Kirby\Toolkit\A;
+use Kirby\Toolkit\F;
+
+trait HasLock
+{
+
+    public function lock($user = false, string $language = null)
+    {
+        // read
+        if ($user === false) {
+            $data = $this->content($language)->get('lock');
+
+            if ($data->isEmpty()) {
+                return ['lock' => false];
+            }
+
+            $data = $data->yaml();
+
+            return array_merge($data, [
+                'lock'  => true,
+                'email' => $this->kirby()->user($data['user'])->email(),
+            ]);
+        }
+
+        // write: unset
+        if ($user === null) {
+            return $this->update([
+                'lock' => null
+            ], $language);
+        }
+
+        // write: set
+        return $this->update([
+            'lock' => [
+                'user' => $user,
```

Comment:
There's no type validation on this. Fixes itself if the method only accepts `string?` (see above).